iroh-gateway/src/handlers.rs

```diff
@@ -42,15 +44,15 @@ use crate::{
     error::GatewayError,
     headers::*,
     response::{get_response_format, GatewayResponse, ResponseFormat},
+    templates::{icon_class_name, ICONS_STYLESHEET, STYLESHEET},
 };
 
 /// Trait describing what needs to be accessed on the configuration
 /// from the shared state.
 pub trait StateConfig: std::fmt::Debug + Sync + Send {
     fn rpc_client(&self) -> &iroh_rpc_client::Config;
-
+    fn public_url_base(&self) -> String;
```

Comment:
Should return `&str` to avoid allocations when not needed, like the other methods on this trait.